rotkehlchen/chain/substrate/typing.py

```diff
@@ -0,0 +1,108 @@
+from enum import Enum
+from typing import Any, Dict, List, NamedTuple, NewType, Tuple, Type, Union
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.fval import FVal
+
+ChainId = NewType('ChainId', str)
+BlockNumber = NewType('BlockNumber', int)
+
+
+class KusamaNodeName(Enum):
+    """
+    TODO: this requires to be extended with OWN_NODE.
+    """
+    PARITY = 0
+    WEB3_FOUNDATION = 1
+
+    def __str__(self) -> str:
+        if self == KusamaNodeName.PARITY:
+            return 'parity'
+        if self == KusamaNodeName.WEB3_FOUNDATION:
+            return 'web3 foundation'
+        raise AssertionError(f'Unexpected KusamaNodeName: {self}')
+
+    def endpoint(self) -> str:
+        if self == KusamaNodeName.PARITY:
+            return 'wss://kusama-rpc.polkadot.io/'
+        if self == KusamaNodeName.WEB3_FOUNDATION:
+            return 'wss://cc3-5.kusama.network/'
+        raise AssertionError(f'Unexpected KusamaNodeName: {self}')
+
+    def is_open(self) -> bool:
+        if self == KusamaNodeName.PARITY:
+            return True
+        if self == KusamaNodeName.WEB3_FOUNDATION:
+            return True
+        raise AssertionError(f'Unexpected KusamaNodeName: {self}')
+
+
+NodeName = Union[KusamaNodeName]
+
+
+class ChainProperties(NamedTuple):
+    """These properties are populated straight from the blockchain
+    """
+    ss58_format: int
+    token: Asset  # from 'tokenSymbol'
+    token_decimals: FVal
+
+
+class SubstrateInterfaceAttributes(NamedTuple):
+    type_registry_preset: str
+
+
+class Chain(Enum):
+    """Supported Substrate chains
+    """
+    KUSAMA = 1
+
+    def __str__(self) -> ChainId:
+        """Return the official chain identifier/name.
+        """
+        if self == Chain.KUSAMA:
+            return ChainId('Kusama')
+        raise AssertionError(f'Unexpected Chain: {self}')
+
+    def get_chain_api(self) -> str:
+        """Return the associated explorer API associated.
+
+        NB: this simplified implementation relies on Subscan API supporting all
+        the chains we introduce.
+        """
+        if self == Chain.KUSAMA:
+            return 'https://kusama.subscan.io/api'
+        raise AssertionError(f'Unexpected Chain: {self}')
+
+    def get_substrate_interface_attributes(self) -> SubstrateInterfaceAttributes:
+        """Return the attributes for instantiating SubstrateInterface.
+        """
+        if self == Chain.KUSAMA:
+            return SubstrateInterfaceAttributes(type_registry_preset='kusama')
+        raise AssertionError(f'Unexpected Chain: {self}')
+
+    def get_blocks_threshold(self) -> BlockNumber:
+        """Return the blocks difference that marks a node as unsynced.
+        """
+        if self == Chain.KUSAMA:
+            return BlockNumber(10)
+        raise AssertionError(f'Unexpected Chain: {self}')
+
+    def get_node_name(self) -> Union[Type[KusamaNodeName]]:
```

Comment:
I had to read it quite a few times to understand this is a type. Please rename to `get_node_name_type`